Gordon360/Services/MembershipService.cs

```diff
@@ -444,5 +390,55 @@ public class ParticipationType
             // BUT, it's convenient to treat it as a participation type in several places throughout the API
             public static ParticipationType GroupAdmin { get { return new ParticipationType("GRP_ADMIN"); } }
         }
+
+        /// <summary>	
+        /// Converts a MembershipUploadViewModel object to a MEMBERSHIP
+        /// </summary>
+        /// <param name="membership">The MembershipUploadViewModel to convert</param>	
+        /// <returns>A new MEMBERSHIP object filled with the info from the view model</returns>	
+        public MEMBERSHIP MembershipUploadToMEMBERSHIP(MembershipUploadViewModel membership)
+        {
+            var sessionCode = _context.CM_SESSION_MSTR
+                .Where(x => x.SESS_CDE.Equals(membership.SessCode))
+                .FirstOrDefault();
+
+            return new MEMBERSHIP() {
+                ACT_CDE = membership.ACTCode,
+                SESS_CDE = membership.SessCode,
+                ID_NUM = int.Parse(_accountService.GetAccountByUsername(membership.Username).GordonID),
+                BEGIN_DTE = (DateTime?)sessionCode?.SESS_BEGN_DTE ?? DateTime.Now,
+                PART_CDE = membership.PartCode,
+                COMMENT_TXT = membership.CommentText,
+                GRP_ADMIN = membership.GroupAdmin,
+                PRIVACY = membership.Privacy,
+                USER_NAME = Environment.UserName
+            };
+        }
+
+        /// <summary>	
+        /// Finds the matching MembershipView object from an existing MEMBERSHIP object
+        /// </summary>
+        /// <param name="membership">The MEMBERSHIP to match on MembershipID</param>	
+        /// <returns>The found MembershipView object corresponding to the MEMBERSHIP by ID</returns>	
+        public MembershipView MEMBERSHIPToMembershipView(MEMBERSHIP membership)
+        {
+            var foundMembership = _context.MembershipView.FirstOrDefault(m => m.MembershipID == membership.MEMBERSHIP_ID);
+
+            if (foundMembership == null)
+            {
+                throw new ResourceNotFoundException();
+            }
+
+            return foundMembership;
+        }
```

Comment:
Move this to a helper class perhaps? @EjPlatzer thoughts?

Reply:
Instead of a helper class, why not make it an extension method?

Reply:
Extensions do not work with implicit operators or static methods. I just renamed it and left it here.